documentation/smg/16_serialiser.md

```diff
@@ -0,0 +1,244 @@
+# The serialiser
+
+In order to efficiently transfer MATLAB information between storage, be it via networked interaction (such as MPI) or to disc efficiently, the information is first "serialised" into a representation. This document details the operation of the serialisers defined in:
+```
+<HERBERT_ROOT>/herbert_core/utilities/misc/hlp_serialise
+<HERBERT_ROOT>/herbert_core/utilities/misc/hlp_deserialise
+<HERBERT_ROOT>/_LowLevelCode/cpp/cpp_communicator/serialiser/c_serialise
+<HERBERT_ROOT>/_LowLevelCode/cpp/cpp_communicator/serialiser/c_deserialise
+```
+
+The serialised format has recently been updated to be more consistent and to have greater capabilities such as serialising arrays of structs/objects.
+
+# New format
+
+## Standard header format
+
+The new style has a standardised header format depending on the shape of the data (with the exception of function handles).
+This format allows extension to support arrays of objects and structs in a way which the old serialiser couldn't.
+It also greatly simplifies the tag structure and allows it to align with the C++ MEX API data types.
+Data is serialised in the linearised order as defined by `array(:)` and the array is subsequently reshaped by the deserialiser to the shape as defined by the header.
```

Comment:
@oerc0122 Hopefully last item. I still don't know what you mean by "as defined by `array(:)`". Do you mean a specific variable 'array', or do you mean the generic unpacking of the MATLAB array that represents data. Or something else. Either way, saying what 'array' refers to. Thanks.